File: ゼミ用/05製造/api/src/request/LoginRequest.php
<?php
require_once dirname(__FILE__) . '/../common/Constant.php';

/**
 * アカウント登録処理の値を保持する
 */
class LoginRequest {
    private $userId;
    private $password;
    private $errorMsg = array();

    /**
     * アカウント登録処理の値を格納します
     */
    public function __construct($userId, $password) {
        $this->userId = $userId;
        $this->password = $password;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function getPassword() {
        return $this->password;
    }

    public function getErrorMsg() {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation() {
        $validationFlg = false;

        // ユーザ名の文字数チェック
        if (strlen($this->userId) != Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR)) {
            array_push($this->errorMsg, 'エラー ：ユーザIDは8文字で入力してください');
            $validationFlg = true;
        }

        // パスワードの文字数チェック
        if (20 < strlen($this->password)) {
            array_push($this->errorMsg, 'エラー ：パスワードは20文字以内で入力してください');
            $validationFlg = true;
        }

        // 入力項目チェック
        if (empty($this->userId) || empty($this->password)) {
            array_push($this->errorMsg, 'エラー ：ユーザIDまたはパスワードが入力されていません');
            $validationFlg = true;
        }

        return $validationFlg;
    }
}

File: ゼミ用/05製造/eeldev/eelPy/api/request/LoginRequest.py
from ..common.Constant import Constant


# /**
# * アカウント登録処理の値を保持する
# */
class LoginRequest:
    errorMsg = []

    # /**
    # * アカウント登録処理の値を格納します
    # */
    def __init__(self, userId, password):
        self.userId = userId
        self.password = password

    def getUserId(self):
        return self.userId

    def getPassword(self):
        return self.password

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # 入力項目チェック
            if (len(self.userId) == 0 or len(self.password) == 0):
                self.errorMsg.append('エラー ：ユーザIDまたはパスワードが入力されていません')
                validationFlg = True

            # ユーザIDの文字数チェック
            if (Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) < len(self.userId)):
                self.errorMsg.append(
                    'エラー ：ユーザIDは' + Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) + '文字以内で入力してください')
                validationFlg = True

            # パスワードの文字数チェック
            if (20 < len(self.password)):
                self.errorMsg.append('エラー ：パスワードは20文字以内で入力してください')
                validationFlg = True      

        except Exception as e:
            validationFlg = True

        return validationFlg
